torch/distributed/_functional_collectives.py

```diff
@@ -244,6 +248,18 @@ def _expand_group(group: RANK_TYPES, tag: str = "") -> Tuple[str, List[int], int
 
     return (tag, rankset, group_size)
 
+def _are_we_tracing() -> bool:
+    mode = get_innermost_proxy_mode()
+    if mode is None:
+        return False
+    return mode.tracer is not None
```

Comment:
Did you intentionally check mode.tracer? It seems to me that this is guaranteed to be not None